dev/breeze/doc/adr/0003-bootstraping-virtual-environment.md

```diff
@@ -0,0 +1,200 @@
+of the CI is stored in the same repository as the source code of the
+application it tests and part of the Breeze functions are shared with CI.
+
+In the future when Breeze2 stabilizes and it's update cadence will be
+much slower (which is likele as it happened with the Breeze predecessor)
+there could be an option that Breeze is installed as separate package and
+same released Breeze version could be ued to manage multiple Airflow
+versions, for that we might want to release Breeze as a separate package
+in PyPI. However since there is the CI integration, the source code
+version of Breeze will remain as part of the Airflow's source code.
+
+
+## Decision
+
+The decision is to implement Breeze in a subfolder (`dev/breeze2/`) of
+Apache Airflow as a Python project following the standard setuptools
+enabled project. The project contains setup.py and dependencies described
+in setup.cfg and contains both source code and tests for Breeze code.
+
+The sub-project could be used in the future to produce a  PyPI package
+(we reserved such package in PyPI), however its main purpose is
+to install Breeze in a separate virtualenv bootstrapped
+automatically in editable mode.
+
+There are two ways you will be able to install `Breeze2` - locally in
+repository using ./Breeze2 bootstrapping script and using `pipx`.
+
+The bootstrapping Python script (`Breeze2` in the main repository
+of Airflow performs the following tasks:
+
+* when run for the first time it creates `.build/breeze2/venv` virtual
+  environment (Python3.6+ based) - with locally installed `dev`
+  project in editable mode (`pip install -e .`) - this makes sure
+  that the users of Breeze will use the latest version of Breeze
+  available in their version of the repository
+* when run subsequently, it will check if setup files changed for
+  Breeze (dependencies changed) and if they did it will automatically
+  reinstall the environment, adding missing dependencies
+* after managing the venv, the Breeze2 script will simply execute
+  the actual Breeze2 script in the `.build/venv` passing the
+  parameters to the script. For the user, the effect will be same
+  as activating the virtualenv and executing the ./Breeze2 from
+  there (but it will happen automatically and invisibly for the
+  user
+* In Windows environment where you have no easy/popular mechanism
+  of running scripts with shebang (#!) equivalent in Posix
+  environments, Users will be able to locally build (using
+  `pyinstaller` a `Breeze2.exe` frozen Python script that will
+  essentially do the same, they could also use `python Breeze2`
+  command or switch to Git Bash to utilize the shebang feature
+  (Git Bash comes together with Git when installed on Windows)
+* The second option is to use `pipx` to install Breeze2.
+  `Pipx` is almost equivalent to what the Bootstrapping does
+  and many users might actually choose to install Breeze this
+  way - and we will add is as an option to install Breeze
```

Comment:
Another typo: we will add `is` as an option... instead of we will add `it` as an option...